### PR TITLE
feat(web-analytics): Add warning when conversion goal is a custom action with no session id

### LINKED ISSUE #26428: Web analytics conversion sometimes don't show in the overview query
## Bug description

The overview query can sometimes show 0 unique and total conversions, but the graph below shows conversions. These 2 queries should match

This was reported here https://posthoghelp.zendesk.com/agent/tickets/21134

## How to reproduce

This doesn't seem to affect the posthog org, but it has affected some customers with these steps to reproduce:

1. Open web analytics
2. Add a conversion goals
3. Observe that total and unique conversions are zero

### Debug info

```shell
- [ ] PostHog Cloud, Debug information: [please copy/paste from https://us.posthog.com/settings/project-details#variables or https://eu.posthog.com/settings/project-details#variables]
- [ ] PostHog Hobby self-hosted with `docker compose`, version/commit: [please provide]
- [ ] PostHog self-hosted with Kubernetes (deprecated, see [`Sunsetting Kubernetes support`](https://posthog.com/blog/sunsetting-helm-support-posthog)), version/commit: [please provide]
```

### PR DESCRIPTION
**Pull request by robbie-c**
## Problem

Events with no session ids don't work in web analytics, and with conversion goals, sometimes people set custom events that are only seen from their server SDK. If they haven't set up session IDs, these will fail.

Mitigates https://github.com/PostHog/posthog/issues/26428

## Changes

Add a warning if one of these events is chosen as the session id

## Does this work well for both Cloud and self-hosted?

Yes

## How did you test this code?

Created a server side event on my local dev set up for testing

### CODE REVIEW

File: frontend/src/scenes/web-analytics/webAnalyticsLogic.tsx [hunk 1]
```diff
@@ -1481,13 +1493,45 @@ export const webAnalyticsLogic = kea<webAnalyticsLogicType>([
...
+            }
+
+            if (!conversionGoal || !('customEventName' in conversionGoal) || !conversionGoal.customEventName) {
+                actions.setConversionGoalWarning(null)
+                return
+            }
+            const { customEventName } = conversionGoal
+            // check if we have any conversion events from the last week without sessions ids
+
+            const response = await hogqlQuery(
```
Comment (rafaeelaudibert) [inline frontend/src/scenes/web-analytics/webAnalyticsLogic.tsx:1512]: Weird blank line between the comment and the query

File: frontend/src/scenes/web-analytics/webAnalyticsLogic.tsx [hunk 2]
```diff
@@ -1481,13 +1493,45 @@ export const webAnalyticsLogic = kea<webAnalyticsLogicType>([
...
+                `select count() from events where timestamp >= now() - toIntervalHour(24) AND ($sessionId IS NULL OR $sessionId = '') AND event = {event}`,
+                { event: customEventName }
+            )
+            breakpoint()
```
Comment (rafaeelaudibert) [inline frontend/src/scenes/web-analytics/webAnalyticsLogic.tsx:1516]: What are these for? I've seem them sprinkled around

File: frontend/src/scenes/web-analytics/webAnalyticsLogic.tsx [hunk 3]
```diff
@@ -268,6 +273,7 @@ export const webAnalyticsLogic = kea<webAnalyticsLogicType>([
         openAsNewInsight: (tileId: TileId, tabId?: string) => {
             return { tileId, tabId }
         },
+        setConversionGoalWarning: (warning: ConversionGoalWarning | null) => ({ warning }),
```
Comment (rafaeelaudibert) [inline frontend/src/scenes/web-analytics/webAnalyticsLogic.tsx:276]: This is my lack of knowledge on kea showing up: why is it that we're using an object here, but then on line 466 we are doing it differently?

My guess is that whatever we pass in here is actually what we get as the second parameter on line 466. Is that right? What's the rationale here?

And yes, I know the answer is probably: Rafa, you have to learn Kea.

File: frontend/src/scenes/web-analytics/webAnalyticsLogic.tsx [hunk 4]
```diff
@@ -1481,13 +1493,45 @@ export const webAnalyticsLogic = kea<webAnalyticsLogicType>([
...
+
+            if (!conversionGoal || !('customEventName' in conversionGoal) || !conversionGoal.customEventName) {
+                actions.setConversionGoalWarning(null)
+                return
+            }
+            const { customEventName } = conversionGoal
+            // check if we have any conversion events from the last week without sessions ids
+
+            const response = await hogqlQuery(
+                `select count() from events where timestamp >= now() - toIntervalHour(24) AND ($sessionId IS NULL OR $sessionId = '') AND event = {event}`,
```
Comment (rafaeelaudibert) [inline frontend/src/scenes/web-analytics/webAnalyticsLogic.tsx:1513]: When is it preferable to run a query inline like this versus outsourcing it to the `<Query/>` component?

**Review by rafaeelaudibert — APPROVED** [review]
Code looks good, even though I'm not really sure if it's correct. The logic looks good though :)

File: frontend/src/scenes/web-analytics/webAnalyticsLogic.tsx (hunk 2 above)
Comment (robbie-c) [reply frontend/src/scenes/web-analytics/webAnalyticsLogic.tsx:1516]: https://keajs.org/docs/core/listeners/#breakpoints these guys :)

In this case it means that if the action has happened while the query was executing, stop here before dispatching that actions on the next lines

**Pull request merged**